### PR TITLE
Warn on low disk space during checkpointing

### PR DESCRIPTION
**Pull request by finbarrtimbers**
## Summary
- add disk space usage check before checkpointing to warn and notify via Slack
- introduce reusable Slack messaging helper and tests

## Testing
- uv run pytest open_instruct/test_utils.py -k slack *(fails: network dependency download error for torch wheel)*

------
[Codex Task](https://chatgpt.com/codex/tasks/task_b_693146434300832da36e11823f082416)

---

> [!NOTE]
> Adds a local disk-usage check before checkpointing that logs and can Slack-notify, and replaces error-only Slack alert with a generic `send_slack_message`; includes focused tests.
> 
> - **Training/runtime (`open_instruct/grpo_fast.py`)**
>   - Add `warn_if_low_disk_space(path, threshold, send_slack_alerts)` with cloud-path bypass and Slack notification support.
>   - Invoke disk check before saving checkpoint state using `DISK_USAGE_WARNING_THRESHOLD`.
>   - Replace failure alert with `utils.send_slack_message` including error details.
> - **Utils (`open_instruct/utils.py`)**
>   - Replace `send_slack_alert(Exception)` with generic `send_slack_message(str)` that appends Beaker URL and handles network errors.
> - **Tests (`open_instruct/test_utils.py`)**
>   - Update Slack tests to use `send_slack_message` and validate message content/Beaker URL.
>   - Add comprehensive tests for `warn_if_low_disk_space` (cloud-path skip, thresholds, Slack path, zero-total).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit d86decf5752aa8791c35b02bdc17191b4cfeceb5. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: open_instruct/grpo_fast.py [hunk 1]
```diff
@@ -132,6 +132,37 @@
 logger = logger_utils.setup_logger(__name__)
 
 INVALID_LOGPROB = 1.0
+DISK_USAGE_WARNING_THRESHOLD = 0.85
+CLOUD_PATH_PREFIXES = ("gs://", "s3://", "az://", "hdfs://")
```
Comment (finbarrtimbers) [inline open_instruct/grpo_fast.py:136]: Are these dumb, as we haven't actually run on these clouds?

Comment (hamishivi) [reply open_instruct/grpo_fast.py:136]: I think this is fine, being overly cautious.
